modules/core/src/main/java/org/apache/ignite/spi/deployment/local/LocalDeploymentSpi.java

```diff
@@ -111,35 +110,60 @@
     }
 
     /** {@inheritDoc} */
-    @Nullable @Override public DeploymentResource findResource(String rsrcName) {
+    @Nullable @Override public DeploymentResource findResource(String rsrcName, @Nullable ClassLoader clsLdr) {
         assert rsrcName != null;
 
-        // Last updated class loader has highest priority in search.
-        for (Entry<ClassLoader, ConcurrentMap<String, String>> e : ldrRsrcs.descendingEntrySet()) {
+        if (clsLdr != null) {
+            ConcurrentMap<String, String> rsrcs = ldrRsrcs.get(clsLdr);
+
+            if (rsrcs == null)
+                return null;
+
+            return findResource0(rsrcs, rsrcName, clsLdr);
+        }
+
+        for (Entry<ClassLoader, ConcurrentMap<String, String>> e : ldrRsrcs.entrySet()) {
```

Comment:
It is a very suspected way, because you cannot be sure what the specific class loader will be chosen.
Is it only for deprecated functionality?
If yest, please extract it to another method and mark it @Deperecated.

Reply:
the same as here https://github.com/apache/ignite/pull/8759#discussion_r572667758